src/game/rooms.js

```diff
@@ -968,6 +968,9 @@ exports.make = function(_runtimeData, _intents, _register, _globals) {
         if(!C.CONSTRUCTION_COST[structureType]) {
             return C.ERR_INVALID_ARGS;
         }
+        if(structureType == 'spawn' && typeof name == 'string' && createdSpawnNames.indexOf(name) != -1) {
```

Comment:
I think you need to check whether the name is already used by another spawn of spawn construction site here, just like you do below when generating a new name:
```_.any(register.spawns, {name}) ||
                _.any(register.constructionSites, {structureType: 'spawn', name})```